### ISSUE TITLE
[rllib]In multiagent environment, is there a way to use common experience to update the policies with different hyperparameters?

### ISSUE THREAD

**Issue by 51616** [ray-project/ray#9743]
As discussed in #8452 , in [PBT-MARL](https://arxiv.org/pdf/1902.07151.pdf) algorithm we need agents/policies with different hyperparameters (e.g. learning rate, gamma, vf_coeff) to play the game together and use the common experience to update both agents' policies. I wonder if this is possible the fact that it requires to use the __same experiences__ to update policies with __different hyperparameters__ in RLlib? 

Edit1. If I want to have different learning rates or batch sizes for each policy, Is it possible to have individual optimizer for each policy in a single trainer instance? or a workaround to achieve this?

This is psuedo-code for the algorithm.
![image](https://user-images.githubusercontent.com/20886355/88617603-4732d280-d0c1-11ea-8e0f-66c82daeb2ca.png)

**Comment by sven1977** [comment]
Yeah, in the MA case, you can configure each policy (in the same trainer) differently.
In your multiagent config, you can specify the `policies` dict via `keys=policy_id[str]` and `values=Tuple([PolicyClass or None for default], obs_space, action_space, [extra config, e.g. different learning rate per policy])`
e.g.
```
config:
    multiagent:
        policies: {
            "pol0": (None, obs-space, action-space, {"lr": 0.001}),
            "pol1": (None, obs-space, action-space, {"lr": 0.002}),
        }
```

**Comment by 51616** [comment]
> Yeah, in the MA case, you can configure each policy (in the same trainer) differently.
> In your multiagent config, you can specify the `policies` dict via `keys=policy_id[str]` and `values=Tuple([PolicyClass or None for default], obs_space, action_space, [extra config, e.g. different learning rate per policy])`
> e.g.
> 
> ```
> config:
>     multiagent:
>         policies: {
>             "pol0": (None, obs-space, action-space, {"lr": 0.001}),
>             "pol1": (None, obs-space, action-space, {"lr": 0.002}),
>         }
> ```
Is this applicable to other configs such as callbacks or this is only for algo specific config?
Also, could you explain a bit more about how this is handled by the trainer because I looked into the source code and I saw only one optimizer for each trainer instance?
